### ISSUE TITLE
[typings] OutputCachedSelector and OutputParametricCachedSelector do not include selectorCreator

### ISSUE THREAD

**Issue by Tyrame** [toomuchdesign/re-reselect#20]
Hi, and thanks for this handy library.

It looks like the `selectorCreator` parameter is missing from the typings.

API Documentation:
```
reReselect([reselect's createSelector arguments])(resolverFunction, selectorCreator = selectorCreator)
```
(This should maybe be replaced with createCachedSelector in the doc to avoid confusion)

Typing:
```typescript
export default function createCachedSelector<S, R1, T> (
  selector: Selector<S, R1>,
  combiner: (res: R1) => T
): OutputCachedSelector<S, T, (res: R1) => T>;

export type OutputCachedSelector<S, R, C> = (resolver: Resolver<S>) => OutputSelector<S, R, C> & {
  getMatchingSelector: (state: S, ...args: any[]) => OutputSelector<S, R, C>;
  removeMatchingSelector: (state: S, ...args: any[]) => void;
  clearCache: () => void;
  resultFunc: C;
};
```

Thanks again, and tell me if I am missing something.

**Comment by toomuchdesign** [comment]
Just pushed an update in [typescript-definitions-selectorcreator](https://github.com/toomuchdesign/re-reselect/tree/fix/typescript-definitions-selectorcreator) branch.

I had to write a very generic typing description in order to not duplicate the whole original `reselect` typings. I still haven't find a way to import TS typing descriptions from another library.

Any further suggestion, @Tyrame?
Thank you!

**Comment by Tyrame** [comment]
In reselect's typings, they have this definition:
```typescript
export function createStructuredSelector<S, T>(
  selectors: {[K in keyof T]: Selector<S, T[K]>},
  selectorCreator?: typeof createSelector,
): Selector<S, T>;
```

I am not an expert, but I think the following code would be ok:
```typescript
export type CreateSelectorInstance = typeof createCachedSelector;
```

**Comment by xsburg** [comment]
@toomuchdesign, I've just pushed a fix that reuses reselect definitions to the same branch, please have a look: https://github.com/toomuchdesign/re-reselect/commit/8ceb27149655e3ebb4182ec630ed0200b89e61dd

**Comment by toomuchdesign** [comment]
@xsburg, It looks too good to be true! Thank you. I'm going to merge ASAP.
We are making open source fun again ™️ ! 😄

PS. Welcome on board as an official collaborator, @xsburg!

**Comment by toomuchdesign** [comment]
Solved by [v0.6.3](https://github.com/toomuchdesign/re-reselect/releases/tag/v0.6.3).
